File: connectx-backend/src/main/java/com/connectx/common/response/ErrorResponse.java
package com.connectx.common.response;

import java.time.Instant;

public class ErrorResponse {

    private String timestamp;
    private int status;
    private String code;
    private String message;
    private String path;

    public ErrorResponse() {
        this.timestamp = Instant.now().toString();
    }

    public ErrorResponse(int status, String code, String message, String path) {
        this.timestamp = Instant.now().toString();
        this.status = status;
        this.code = code;
        this.message = message;
        this.path = path;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/core/network/model/ErrorResponse.kt
package com.connectx.app.core.network.model

import kotlinx.serialization.Serializable

/**
 * Mirrors the backend's com.connectx.common.response.ErrorResponse exactly
 * (connectx-backend/src/main/java/com/connectx/common/response/ErrorResponse.java),
 * as returned by GlobalExceptionHandler on every non-2xx response. This is a
 * contract-only model for N2.1 -- it is not yet wired into any error-parsing
 * logic (that belongs to N2.5).
 */
@Serializable
data class ErrorResponse(
    val timestamp: String? = null,
    val status: Int = 0,
    val code: String? = null,
    val message: String? = null,
    val path: String? = null
)
